packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import {
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+} from '../index';
+
+type PolymorphicActionOwnProps = {
+  /**
+   * Content in the action element.
+   */
+  children?: React.ReactNode;
+};
+
+export type PolymorphicActionProps = PolymorphicComponentProps<
+  'a',
+  PolymorphicActionOwnProps
+>;
+
+/**
+ * Polymorphic action component.
+ * It is rendered as `a` by default.
+ *
+ */
+export const PolymorphicAction = React.forwardRef((props, ref) => {
```

Comment:
Need to think about the naming. `PolymorphicAction` is too low-level, and even the JSDoc explains nothing.

Maybe you could list out any other names that you can think of, and do some research into other libraries? One that I've found is `LinkBox` + `LinkOverlay` from chakra-ui. 
https://chakra-ui.com/docs/components/link-overlay

Reply:
`ActionOverlay`? I don't want to blindly copy from other libraries. Maybe we need team vote?

Reply:
Added two components. LinkBox has tabIndex to skip it when tabbing.

Reply:
The naming is kinda reversed. Box should the outer wrapper, and Overlay should be inner one.

And why do you want to skip it when tabbing?

Reply:
Reversed the naming.

In the article linked in the issue, when you tab through the tiles it tabs only once per tile. I think it's more smooth behavior - click on full tile to open a link, not same action in two places. I could be wrong.

Reply:
Removed tabIndex from Tile, should be good now.

Reply:
I think everything from this comment is solved.

Reply:
this comment is about naming. Have you tried to research into what other libraries are calling it and/or come up with other names yourself?

Reply:
LinkBox for wrapper and LinkAction for action?
I have almost no ideas on naming. Did not find anywhere else besides chakra-ui.

Reply:
> LinkBox for wrapper and LinkAction for action?
i like it. sounds more intuitive than "overlay".